gen/eth_idma_reg.hjson

```diff
@@ -0,0 +1,294 @@
+{
+    name: "eth_idma",
+    clock_primary: "msoc_clk",
+    reset_primary: "rst_int",
+    bus_interfaces: [
+        { protocol: "reg_iface", direction: "device"}
+    ]
+    regwidth: "32",
+    registers: [
+        { name: "maclo_addr",
+          desc: "lower 4 bytes of the devices MAC address",
+          swaccess: "rw",
+          hwaccess: "hro",
+          fields: [
+              { bits: "31:0", name: "lower_mac_address",
+                desc: "Lower 32 bit of the devices MAC address"
+                resval: "8980226"
+              }
+          ]
+        },
+        { name: "machi_mdio",
+          desc: "upper 2 bytes of the devices MAC address, promiscuous flag, MDIO interface",
+          swaccess: "rw",
+          hwaccess: "hro",
+          fields: [
+              { bits: "15:0", name: "upper_mac_address",
+                desc: "Upper 16 bit of the devices MAC address"
+                resval: "8961"
+              },
+              { bits: "16", name: "promiscuous",
+                desc: "promiscuous flag"
+              },
+              { bits: "17", name: "phy_mdclk",
+                desc: "MDIO clock"
+              },
+              { bits: "18", name: "phy_mdio_o",
+                desc: "MDIO output"
+              },
+              { bits: "19", name: "phy_mdio_oe",
+                desc: "MDIO output enable"
+              }
+          ]
+        },
+        { name: "tx_fcs",
+          desc: "The FCS TX status",
+          swaccess: "ro",
+          hwaccess: "hwo",
+          fields: [
+              { bits: "31:0", name: "tx_fcs_reg",
+                desc: "FCS TX status"
+              }
+          ]
+        },
+        { name: "rx_fcs",
+          desc: "The FCS RX status",
+          swaccess: "ro",
+          hwaccess: "hwo",
+          fields: [
+              { bits: "31:0", name: "rx_fcs_reg",
+                desc: "FCS RX status"
+              }
+          ]
+        },
+        { name: "src_addr",
```

Comment:
I would prepend idma-specific registers with `idma_`

Also, design question to be discussed with @thommythomaso 

AFAIU, this register file for ethernet-idma contains also the registers of the 'idma' part. 

Since the idma already comes with its register file in different flavors, should we reuse that, and add the ethernet-specific registers as separate RF?

What do you think?

Reply:
Replying to this before you change something: let's keep things like now (1 register file) :) in the future we can decouple and reuse the RF, now let's focus on functionality. I just wanted to point it out

Reply:
sure. thanks!